Handling network call with Coroutines

igor1797 opened the pull request:
Added Coroutines usage for network call.

igor1797 committed:
Changed sealed kotlin class usage.

igor1797 committed:
Changed sealed kotlin class usage.

igor1797 committed:
Changed TabLayoutMediator usage.

igor1797 committed:
Changed ViewBinding and DataBinding usage.

igor1797 committed:
Removed ViewBinding from build.gradle.

igor1797 committed:
Added TextInputLayout.

igor1797 committed:
Added Coroutines usage for network call.

igor1797 committed:
Added DiffUtil usage for displaying lists of repositories and users.

bagy94 committed:
Merge pull request #10 from igor1797/enhancement/sealed-kotlin-class-usage

Changed sealed kotlin class usage

bagy94 committed:
Merge pull request #15 from igor1797/enhancement/diff-util-usage

Added DiffUtil usage.

bagy94 commented on app/src/main/java/igor/kuridza/dice/githubapp/ui/fragments/repositories/RepositoriesListViewModel.kt:
```diff
@@ -16,7 +18,21 @@ class RepositoriesListViewModel(
 
-    fun getRepositoriesVyQuery(query: String){
-        mRepositoriesList = searchRepository.getRepositoriesBySearchQuery(query) as MutableLiveData<Resource<List<Repository>>>
+    fun getRepositoriesByQuery(query: String) {
```
Jel mozes za ovo napraviti funkciju da nemas ponavljanja logike? Istu stvar radis i sa userima, samo drugu rutu pozivas.

bagy94 committed:
Merge pull request #13 from igor1797/invalid/text-input-layout-usage

Added TextInputLayout usage

bagy94 committed:
Merge pull request #12 from igor1797/bug/view-binding-and-data-binding-usage

Changed ViewBinding and DataBinding usage.

igor1797 committed:
Removed duplicate code.

igor1797 committed:
Resolving conflicts

# Conflicts:
#	app/src/main/java/igor/kuridza/dice/githubapp/model/Resource.kt
#	app/src/main/java/igor/kuridza/dice/githubapp/repositories/SearchRepository.kt
#	app/src/main/java/igor/kuridza/dice/githubapp/ui/fragments/repositories/RepositoriesListFragment.kt
#	app/src/main/java/igor/kuridza/dice/githubapp/ui/fragments/repositories/RepositoriesListViewModel.kt
#	app/src/main/java/igor/kuridza/dice/githubapp/ui/fragments/search/searchrepositories/SearchRepositoriesFragment.kt
#	app/src/main/java/igor/kuridza/dice/githubapp/ui/fragments/search/searchusers/SearchUsersFragment.kt
#	app/src/main/java/igor/kuridza/dice/githubapp/ui/fragments/users/UsersListFragment.kt
#	app/src/main/java/igor/kuridza/dice/githubapp/ui/fragments/users/UsersListViewModel.kt

bagy94 commented on app/src/main/java/igor/kuridza/dice/githubapp/ui/fragments/repositories/RepositoriesListFragment.kt:
```diff
@@ -48,15 +48,12 @@ class RepositoriesListFragment : BaseFragment<RepositoriesListFragmentBinding>()
-    private fun observeRepositories(){
-        viewModel.repositoriesList.observe(this){
+        viewModel.getRepositoriesByQuery(query)
+        viewModel.repositories.observe(this){
```
Nije bas najbolja praksa koristiti fragment kao lifecycleOwner prilikom observanja live data, pogotovo ne kada radis postavljanje u View. Fragment ima drugaciji lifecycle, te moze ostati ziv iako mu je view unisten (backstack). Pripaziti koji se lifecycleOwner postavlja (https://medium.com/@cs.ibrahimyilmaz/viewlifecycleowner-vs-this-a8259800367b)

Vidim da se ovo ponavlja na vise mjesta, prodji kroz sve i prepravi.
